src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java

```diff
@@ -131,6 +131,31 @@ protected final Checker createChecker(Configuration moduleConfig)
         return createChecker(moduleConfig, findModuleCreationOption(name));
     }
 
+    /**
+     * Creates {@link Checker} instance for list  of  {@link Configuration} instances
+     * with parent {@link TreeWalker}.
+     *
+     * @param moduleConfig {@link Configuration} instances.
+     * @return {@link Checker} instance.
+     * @throws Exception if an exception occurs during checker configuration.
+     */
+    protected final Checker createChecker(List<Configuration> moduleConfig)
+            throws Exception {
+        final Configuration dc;
+        dc = createTreeWalkerConfig(moduleConfig);
+        final Checker checker = new Checker();
```

Comment:
Yes @rnveach I need to test it to find out whether the thing is working fine or not


> why isn't this `createChecker(createTreeWalkerConfig(moduleConfig))`?

I did this because with the thought in mind that `createChecker(createTreeWalkerConfig(moduleConfig))` is going to call `createTreeWalkerConfig(Configuration config)` and thus  will result in unnecessary function calls.

Reply:
Does it making any sense?

Reply:
You are doing more than `checker.configure`. You are copying pasting a bunch of other code, probably around 10 lins, duplicating the specifics of creating the checker instance including things like locale. If we change the other methods, we have to now remember to change this one too, which isn't always the case.

If the other method does what we want, than we should use it.